src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c

```diff
@@ -679,6 +679,15 @@ static int get_header_from_vsec(struct feature_rom *rom)
 
 	ret = xocl_subdev_vsec(xdev, XOCL_VSEC_UUID_ROM, &bar, &offset, NULL);
 	if (ret) {
+		/* XOCL_DSAFLAG_CUSTOM_DTB is used for non-VSEC platforms which
+		 * still wanted to use partition metadata to discover resources
+		 */
+		if (XDEV(xdev)->priv.flags & XOCL_DSAFLAG_CUSTOM_DTB) {
+			rom->uuid_len = 32;
+			strcpy(rom->uuid,"fedcba987654321fedcba987654321fe");
```

Comment:
Sonal agreed with this right? Please double check with @maxzhen, this is the first time we hardcode uuid inside driver. We are ok with hardcode configurations, but this seems tied with one specific shell, if you upgrade the shell, driver will be broken. Why do we push this code into XRT master?

Reply:
not sure why we want to hard code a uuid?

Reply:
Can this UUID be passed as a module param which will force users to know what they are doing?
Also please add a note in the code above that UUIDs are best obtained via VSEC and this side band is for testing.

Reply:
Hi @houlz0507 / @maxzhen  / @xdavidz, Sent a reply via email about the background of this change.

Sure @sonals , We can add a module param for this. We will also add a comment.

Reply:
@sonals added changes to pass uuid as module param, please review